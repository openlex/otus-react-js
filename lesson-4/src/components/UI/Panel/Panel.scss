.block {
	border-radius: 5px;
	max-width: 700px;
	font-size: 16px;
	box-shadow: 0 2px 12px -3px rgba(0, 0, 0, .3);
	padding: 10px 15px;
}
