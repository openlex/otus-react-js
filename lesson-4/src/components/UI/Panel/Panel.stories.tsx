import * as React from 'react';
import { Panel } from './Panel';
import { withKnobs, text } from '@storybook/addon-knobs';

export default {
	title: 'ui/Панель - контейнер для элементов',
	components: Panel,
	decorators: [withKnobs]
}

export const DefaultPanel = () =>
	<Panel>{ text('Текст', 'Это красивая панель в которую можно вставить контент. Это красивая панель в которую можно вставить контент') }</Panel>;
