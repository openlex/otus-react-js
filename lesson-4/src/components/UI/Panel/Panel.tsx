import React from "react";
import cn from "classnames";
import * as s from "./Panel.scss";

interface IPanelProps {
  className?: string;
  children: React.ReactNode;
}

export const Panel: React.FC<IPanelProps> = (props: IPanelProps) => {
  return <div className={cn(s.block, props.className)}>{props.children}</div>;
};
