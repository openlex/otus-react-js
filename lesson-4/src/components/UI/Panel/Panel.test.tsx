import React from "react";
import { Panel } from "./Panel";
import { shallow } from "enzyme";

describe("Panel item markup", () => {
	it("match Snapshot", () => {
		const panel = shallow(<Panel>Panel content</Panel>);

		expect(panel).toMatchSnapshot();
	});

})
