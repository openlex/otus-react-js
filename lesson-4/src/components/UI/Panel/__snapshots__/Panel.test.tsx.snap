// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel item markup match Snapshot 1`] = `ShallowWrapper {}`;
